components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/IdentityOAuth2ScopeValidationException.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth2;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.base.IdentityException;
+
+/**
+ * OAuth scope validation exception.
+ */
+
+public class IdentityOAuth2ScopeValidationException extends IdentityException {
```

Comment:
Check formatting on the class.

Also think whether we can improve the class name (whether the class name serves the types of exception we handle)

Reply:
@shashimalcse

Reply:
can we use `IdentityOAuth2ScopeException` instead of this new exception